### PR TITLE
Adjust Miri to also require return places everywhere

### PR DESCRIPTION
**Pull request by JakobDegen**
This is the miri side of rust-lang/rust#96098 . It'll still need a bump to rust-version once the rust PR is merged, but the test suite passes against my local build of rustc.

### CODE REVIEW

File: src/diagnostics.rs [hunk 1]
```diff
@@ -275,7 +275,7 @@ pub fn report_error<'tcx, 'mir>(
     for (i, frame) in ecx.active_thread_stack().iter().enumerate() {
         trace!("-------------------");
         trace!("Frame {}", i);
-        trace!("    return: {:?}", frame.return_place.map(|p| *p));
+        trace!("    return: {:?}", frame.return_place);
```
Comment (RalfJung) [inline src/diagnostics.rs:278]: Should this be `*frame.return_place`?
The intention of the `map(|p| *p)` probably was to `Debug`-print a `MemPlace`, not a `MPlaceTy`. The latter are very verbose due the layout they contain.

Comment (JakobDegen) [reply src/diagnostics.rs]: Makes sense, fixed

**Comment by RalfJung** [comment]
https://github.com/rust-lang/rust/pull/96098 just landed; could you rebase and update the rust-version file? Then this is good to go. :)
@bors delegate+

**Comment by JakobDegen** [comment]
Ah, apologies, was asleep. Thanks for handling this!

**Pull request merged**